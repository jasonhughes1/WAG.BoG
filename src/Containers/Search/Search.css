.search {
  width: 200px; }

.button {
  margin-top: 10px; }

.header-container {
  background-image: url("../../Images/caleb-fisher-7320_Fotor.jpg");
  background-size: cover;
  height: 500px;
  display: flex;
  flex-direction: column;
  align-items: center; }

.whos {
  color: white; }

.search-for {
  color: white;
  margin-top: -10px; }

.dog-cards {
  background-image: url("../../Images/jonathan-forage-367660.jpg");
  background-size: cover; }
