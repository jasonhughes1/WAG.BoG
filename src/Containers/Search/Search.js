import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import RootReducers from '../../Reducers/RootReducers';
import { connect } from 'react-redux';
import { fetchDogSuccess, fetchDogs, searchCurrentDogs} from '../../Actions/Actions';
import Card from '../../Components/Card/Card';
import css from './Search.css';
import PropTypes from 'prop-types';

export class Search extends Component {
  constructor() {
    super()
      this.state = {
        searchValue: '',
        location: '12563'
      }
    }

    handleChange(event) {
      this.setState({
        location: event.target.value
      })
    }


  render() {
    const dogCards = this.props.searchForDogs.map((dog, index) => {
      return <Card
        key={index}
        name={dog.name}
        breed={dog.breed}
        sex={dog.sex}
        picture={dog.photo}
        street={dog.street}
        city={dog.city}
        state={dog.state}
      />
    })

    const { searchValue, location } = this.state;
      return(
        <div>
          <div className='header-container'>
          <h1 className='whos'>Who's a good boy, or girl?</h1>
          <h2 className='search-for'>Search for your dog here</h2>
        <input className='search' placeholder='Please search by zip code' onChange={(event) => this.handleChange(event)} />
        <button className='button' onClick={ () =>  this.props.searchCurrentDogs(location) }>Search!</button>
      </div>
        <div className='card-container'>
          <div className='dog-cards'>{dogCards}</div>
        </div>
      </div>
          )
        }
      }



export const mapStateToProps = (store) => {
  return {
    dogs: store.dogs,
    searchForDogs: store.searchForDogs
  }
}

export const mapDispatchToProps = (dispatch) => ({
  searchCurrentDogs: (location) => {
    dispatch(searchCurrentDogs(location))
  }
});



export default connect(mapStateToProps, mapDispatchToProps)(Search);
