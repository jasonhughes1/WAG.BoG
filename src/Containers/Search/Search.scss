
.search {
  width: 200px;
}

.button {
  margin-top: 10px;
}


.header-container {
  background-image: url('../../Images/caleb-fisher-7320.jpg');
  background-size: cover;
  // filter: brightness(30%);
  height: 500px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.whos {

}

.search-for {
  margin-top: -10px;
}

.dog-cards {
  // background-color: #5C3A2B;
  background-image: url('../../Images/jonathan-forage-367660.jpg');
  background-size: cover;
}
