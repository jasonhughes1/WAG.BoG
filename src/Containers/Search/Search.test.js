import React from 'react';
import { shallow } from 'enzyme';
import { Search } from './Search';

describe('Search tests', () => {
  let search;
  beforeEach( () => {
    const mockFunc = jest.fn();
    search = shallow(
      <Search
        dogs= {['pups']}
        history= {{key: 'value'}}
        location= {{key: 'value'}}
        match= {{key: 'value'}}
        searchCurrentDogs= {mockFunc}
        searchForDogs= {['pups']}
      />);
  });


  it('should match the snapshot', () => {
    expect(search).toMatchSnapshot();
  });

  it('Should be defined', () => {
    expect(search).toBeDefined();
  });
});
