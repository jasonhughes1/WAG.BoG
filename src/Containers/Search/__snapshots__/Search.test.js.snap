// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    dogs={
        Array [
            "pups",
          ]
    }
    history={
        Object {
            "key": "value",
          }
    }
    location={
        Object {
            "key": "value",
          }
    }
    match={
        Object {
            "key": "value",
          }
    }
    searchCurrentDogs={[Function]}
    searchForDogs={
        Array [
            "pups",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header-container"
>
          <h1
                    className="whos"
          >
                    Who's a good boy, or girl?
          </h1>
          <h2
                    className="search-for"
          >
                    Search for your dog here
          </h2>
          <input
                    className="search"
                    onChange={[Function]}
                    placeholder="Please search by zip code"
          />
          <button
                    className="button"
                    onClick={[Function]}
          >
                    Search!
          </button>
</div>,
        <div
          className="card-container"
>
          <div
                    className="dog-cards"
          >
                    <Connect(Component)
                              breed={undefined}
                              city={undefined}
                              name={undefined}
                              picture={undefined}
                              sex={undefined}
                              state={undefined}
                              street={undefined}
                    />
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="whos"
>
              Who's a good boy, or girl?
</h1>,
            <h2
              className="search-for"
>
              Search for your dog here
</h2>,
            <input
              className="search"
              onChange={[Function]}
              placeholder="Please search by zip code"
/>,
            <button
              className="button"
              onClick={[Function]}
>
              Search!
</button>,
          ],
          "className": "header-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Who's a good boy, or girl?",
              "className": "whos",
            },
            "ref": null,
            "rendered": "Who's a good boy, or girl?",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search for your dog here",
              "className": "search-for",
            },
            "ref": null,
            "rendered": "Search for your dog here",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search",
              "onChange": [Function],
              "placeholder": "Please search by zip code",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search!",
              "className": "button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Search!",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="dog-cards"
>
            <Connect(Component)
                        breed={undefined}
                        city={undefined}
                        name={undefined}
                        picture={undefined}
                        sex={undefined}
                        state={undefined}
                        street={undefined}
            />
</div>,
          "className": "card-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Component)
                breed={undefined}
                city={undefined}
                name={undefined}
                picture={undefined}
                sex={undefined}
                state={undefined}
                street={undefined}
/>,
            ],
            "className": "dog-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "breed": undefined,
                "city": undefined,
                "name": undefined,
                "picture": undefined,
                "sex": undefined,
                "state": undefined,
                "street": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-container"
>
            <h1
                        className="whos"
            >
                        Who's a good boy, or girl?
            </h1>
            <h2
                        className="search-for"
            >
                        Search for your dog here
            </h2>
            <input
                        className="search"
                        onChange={[Function]}
                        placeholder="Please search by zip code"
            />
            <button
                        className="button"
                        onClick={[Function]}
            >
                        Search!
            </button>
</div>,
          <div
            className="card-container"
>
            <div
                        className="dog-cards"
            >
                        <Connect(Component)
                                    breed={undefined}
                                    city={undefined}
                                    name={undefined}
                                    picture={undefined}
                                    sex={undefined}
                                    state={undefined}
                                    street={undefined}
                        />
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="whos"
>
                Who's a good boy, or girl?
</h1>,
              <h2
                className="search-for"
>
                Search for your dog here
</h2>,
              <input
                className="search"
                onChange={[Function]}
                placeholder="Please search by zip code"
/>,
              <button
                className="button"
                onClick={[Function]}
>
                Search!
</button>,
            ],
            "className": "header-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Who's a good boy, or girl?",
                "className": "whos",
              },
              "ref": null,
              "rendered": "Who's a good boy, or girl?",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search for your dog here",
                "className": "search-for",
              },
              "ref": null,
              "rendered": "Search for your dog here",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search",
                "onChange": [Function],
                "placeholder": "Please search by zip code",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search!",
                "className": "button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Search!",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="dog-cards"
>
              <Connect(Component)
                            breed={undefined}
                            city={undefined}
                            name={undefined}
                            picture={undefined}
                            sex={undefined}
                            state={undefined}
                            street={undefined}
              />
</div>,
            "className": "card-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Component)
                  breed={undefined}
                  city={undefined}
                  name={undefined}
                  picture={undefined}
                  sex={undefined}
                  state={undefined}
                  street={undefined}
/>,
              ],
              "className": "dog-cards",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "breed": undefined,
                  "city": undefined,
                  "name": undefined,
                  "picture": undefined,
                  "sex": undefined,
                  "state": undefined,
                  "street": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
